### ISSUE TITLE
CRC Failing 

### ISSUE THREAD

**Issue by andrew-stuart-connell** [ecree-solarflare/ovenctl#1]
First,  thanks for sharing this software.

My problem is that when the oven mode is read the CRC check fails.

The Binder chamber I am using communicates normally with Binder software.

**Comment by ecree-solarflare** [comment]
Is this the first command sent to the oven, or have other commands succeeded?  Does it fail repeatably on this specific command?
Could you run under a debugger (or with a patched `except` clause) and provide `crc`, `checkcrc` and `msgbytes`?
What happens if you try to read the mode with the `nmbtest` client in the `tools/` directory (`./nmbtest hostname -a1a22 -v`)?

**Comment by andrew-stuart-connell** [comment]
My bad, I was accessing a different chamber  an MKF 56, it is using table MB2.

There was a small bug which caused the CRC problem which occureed in 

def parse_err_response(msgbytes): # string -> (bool, int)

It was:

    crc, = struct.unpack('<h', msgbytes[3:5])

I changed it to:

    crc, = struct.unpack('<H', msgbytes[3:5])

I then received a correct error code and found I was accessing the wrong table for my oven.

I changed to MB2 addresses and now it works.

Thanks for your time.

Andrew
